tests/integration/ha/helpers.py

```diff
@@ -490,18 +504,19 @@ async def continuous_writes_increases(ops_test: OpsTest, unit_ip: str, app: str)
     )
     writer = ContinuousWrites(ops_test, app, initial_count=initial_count)
     await writer.start()
-    time.sleep(5)
+    time.sleep(10)
     result = await writer.stop()
     return result.count > initial_count
 
 
-async def backup_cluster(ops_test: OpsTest, leader_id: int) -> bool:
+async def backup_cluster(ops_test: OpsTest, leader_id: int) -> int:
     """Runs the backup of the cluster."""
     action = await run_action(ops_test, leader_id, "create-backup")
+    assert action.status == "completed"
```

Comment:
Similar to a previous comment (wait_for_backup), it is discouraged to have asserts in non-assert methods, the preferred way is to throw a regular exception. There needs to be a separation between test functions and helpers. 
This can be the object of another PR.

Reply:
@Mehdi-Bendriss done on this PR: https://github.com/canonical/opensearch-operator/blob/53d44ff91f894c48395c5c584bbc751b45f50b6b/tests/integration/ha/helpers.py

Which is based in this here.